File: lib/to_mdx/parsers/open_document/style/master_page.rb
# frozen_string_literal: true

require_relative "../base"

module ToMdx
  module Parsers
    module Style
      class MasterPage < Base
        attr_reader :master_page_name

        def initialize(element = nil, context = {})
          super(element, context)
          @master_page_name = element&.attributes&.[]("style:name")
        end

        def to_mdx
          background_objects = contentful_children.select do |child|
            layer_attr = child.element&.attributes&.[]("draw:layer")
            layer_attr == "backgroundobjects"
          end

          background_objects.map(&:to_mdx).compact.join("\n")
        end

        def to_s
          contentful_children.map(&:to_s).join("\n")
        end

        register_for "style:master-page"
      end
    end
  end
end

File: src/parsers/open-document/style/master_page.ts
import type { ParseContext } from "../../base_element.ts";
import { BaseElement } from "../../base_element.ts";

export class MasterPage extends BaseElement {
  private name: string | null;

  constructor(element: Element, context: ParseContext, parent: BaseElement | null) {
    super(element, context, parent);
    this.name = this.attr("style:name");
  }

  toMdx(): string {
    const backgroundObjects = this.contentfulChildren().filter((child) => {
      return child.attr("draw:layer") === "backgroundobjects";
    });

    return backgroundObjects
      .map((child) => child.toMdx())
      .filter(Boolean)
      .join("\n");
  }

  toString(): string {
    return this.contentfulChildren()
      .map((child) => child.toString())
      .join("\n");
  }

  static {
    BaseElement.registerFor.call(this, "style:master-page");
  }
}
